relay-server/src/services/processor/playstation.rs

```diff
@@ -287,6 +282,31 @@ fn infer_content_type(filename: &str) -> ContentType {
     }
 }
 
+fn create_item_with_json_payload(payload: &str) -> Item {
+    let mut item = Item::new(ItemType::Event);
+    item.set_payload(ContentType::Json, payload.to_owned());
+    item
+}
+
+fn merge_or_create_event_item(json: &str, event: Option<Item>) -> Item {
+    match event {
+        Some(event) => {
+            let event_json =
+                serde_json::from_slice::<serde_json::Value>(&event.payload()).unwrap_or_default();
+            let mut base_json = serde_json::from_str::<serde_json::Value>(json).unwrap_or_default();
```

Comment:
nit: not JSON once parsed.

```suggestion
            let event =
                serde_json::from_slice::<serde_json::Value>(&event.payload()).unwrap_or_default();
            let mut base_event = serde_json::from_str::<serde_json::Value>(json).unwrap_or_default();
```